src/Xamarin.Android.Build.Tasks/Tasks/R8.cs

```diff
@@ -117,6 +119,15 @@ protected override CommandLineBuilder GetCommandLineBuilder ()
 				//NOTE: we may be calling r8 *only* for multi-dex, and all shrinking is disabled
 				cmd.AppendSwitch ("--no-tree-shaking");
 				cmd.AppendSwitch ("--no-minification");
+				// Rules to turn off optimizations
+				var temp = Path.GetTempFileName ();
+				File.WriteAllLines (temp, new [] {
+					"-dontoptimize",
```

Comment:
can I assume that `r8` doesn't care if you have duplicate entries? What does it do about conflicting ones? Could we get in to a situation where a User ha a custom config, which turns on optimisations, but we then turn it off again?

Reply:
In this case (multi-dex only), we don't use any of the user's proguard rules at all.

But there might be an issue when you are using a "Code Shrinker".

In the case `AndroidLinkTool=r8`, we do pass:

    --pg-conf " C:\Program Files (x86)\Android\android-sdk\tools\proguard\proguard-android.txt"

And this file is probably going to trump anything the user supplies. But we have the same code for r8 & proguard:

https://github.com/xamarin/xamarin-android/blob/431329d38137ce47c2d92f76509b84ef8567c33d/src/Xamarin.Android.Build.Tasks/Tasks/Proguard.cs#L127-L135

https://github.com/xamarin/xamarin-android/blob/431329d38137ce47c2d92f76509b84ef8567c33d/src/Xamarin.Android.Build.Tasks/Tasks/R8.cs#L100-L108

In both cases this is first:

https://github.com/xamarin/xamarin-android/blob/431329d38137ce47c2d92f76509b84ef8567c33d/src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets#L319

Another issue is the header of this proguard file:

```
# This is a configuration file for ProGuard.
# http://proguard.sourceforge.net/index.html#manual/usage.html
#
# This file is no longer maintained and is not used by new (2.2+) versions of the
# Android plugin for Gradle. Instead, the Android plugin for Gradle generates the
# default rules at build time and stores them in the build directory.
```

So I should file an issue to look into this at some point.